### ISSUE TITLE
New Octane Rule: `no-classic-classes`

### ISSUE THREAD

**Issue by patocallaghan** [ember-cli/eslint-plugin-ember#539]
Pulled from https://github.com/ember-cli/eslint-plugin-ember/issues/533 which proposes a new set of rules for Octane related best practices

Disallow "classic" classes preferring use of native JS classes

```js
// ### Good
export default class MyComponent extends Component {
  ...
}

// This mixin usage should be okay
export default class MyService extends Service.extends(Evented) {
  ...
}

// ### Bad
export default Component.extend({
  ...
});
```

If anyone would like to implement this please comment that you'll take it and work away.

**Comment by patocallaghan** [comment]
@rwjblue could you sanity check these good/bad examples please?

Is the Mixin usage okay or should that have a lint error too? Should it just be a lint error for everywhere we do `Object.extend`?

**Comment by rwjblue** [comment]
> Is the Mixin usage okay or should that have a lint error too?

IMHO, that should be a separate rule (`no-mixin-usage` or something like that).

**Comment by rwjblue** [comment]
Good/bad examples look good to me.

**Comment by scalvert** [comment]
Ya I thought there was a rule that someone was writing for no mixins? This seems good.

**Comment by alexlafroscia** [comment]
I'm happy to take this one on -- @patocallaghan can you set me as the "assignee"? It's helpful when browsing the issues and seeing which new rules are yet un-claimed!

**Comment by patocallaghan** [comment]
Unfortunately I don’t have admin rights to do that. @bmish could you assign this issue please?
